packages/shared/renderer/sagas.ts

```diff
@@ -295,11 +294,18 @@ function* handleSubmitProfileToRenderer(action: SendProfileToRenderer): any {
       getUnityInstance().LoadProfile(forRenderer)
     }
   } else {
-    const forRenderer = profileToRendererFormat(profile.data, {
-      baseUrl: fetchContentServerWithPrefix
-    })
-    getUnityInstance().AddUserProfileToCatalog(forRenderer)
-    yield put(addedProfileToCatalog(userId, profile.data))
+    const lastSentProfileVersion: number | undefined = yield select(getLastSentProfileVersion, userId)
+
+    // Add version check before submitting profile to renderer
+    if (!lastSentProfileVersion || lastSentProfileVersion < profile.data.version) {
```

Comment:
I don't think it's possible right now because of our current flow where new users are forced to create an avatar at the very beginning, but technically profile version might be `0` and make `!lastSentProfileVersion` always `true`

Reply:
Great comment! Thx! 🙌🏼